code/modules/hydroponics/hydro_tray_process.dm

```diff
@@ -0,0 +1,340 @@
+					if(2)
+						msg_admin_attack("space vines ([seed.display_name]) have spread out of a tray. <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[T.x];Y=[T.y];Z=[T.z]'>(JMP)</a>")
+
+	check_level_sanity()
+	update_icon()
+	return
+
+/obj/machinery/portable_atmospherics/hydroponics/proc/check_health()
+	if(health <= 0)
+		die() //ominous
+
+/obj/machinery/portable_atmospherics/hydroponics/proc/affect_growth(var/amount)
+	if(amount > 0)
+		if(age < seed.maturation)
+			age += amount
+		else if(!harvest && seed.yield != -1)
+			lastproduce -= amount
+	else
+		if(age < seed.maturation)
+			skip_aging++
+		else if(!harvest && seed.yield != -1)
+			lastproduce += amount
+
+
+/obj/machinery/portable_atmospherics/hydroponics/proc/update_name()
+	if(seed)
+		//name = "[initial(name)] ([seed.seed_name])"
+		name = "[seed.display_name]"
+	else
+		name = initial(name)
+
+	if(labeled)
+		name += " ([labeled])"
+
+//Refreshes the icon and sets the luminosity
+/obj/machinery/portable_atmospherics/hydroponics/update_icon()
+	overlays.len = 0
+
+	update_name() //fuck it i'll make it not happen constantly later
+
+	// Updates the plant overlay.
+	if(!isnull(seed))
+		if(draw_warnings && health <= (seed.endurance / 2))
+			overlays += image(seed.plant_dmi,"over_lowhealth3")
+
+		if(dead)
+			overlays += image(seed.plant_dmi,"[seed.plant_icon]-dead")
+		else if(harvest)
+			overlays += image(seed.plant_dmi,"[seed.plant_icon]-harvest")
+		else if(age < seed.maturation)
+			var/t_growthstate = max(1,round((age * seed.growth_stages) / seed.maturation))
+			overlays += image(seed.plant_dmi,"[seed.plant_icon]-grow[t_growthstate]")
+			lastproduce = age
+		else
+			overlays += image(seed.plant_dmi,"[seed.plant_icon]-grow[seed.growth_stages]")
+
+	//Draw the cover.
+	if(closed_system)
+		overlays += "hydrocover"
+
+	//Updated the various alert icons.
+	if(draw_warnings)
+		if(waterlevel <= 10)
+			overlays += "over_lowwater3"
+		if(nutrilevel <= 2)
+			overlays += "over_lownutri3"
+		if(weedlevel >= 5 || pestlevel >= 5 || toxins >= 40 || improper_heat || improper_light || improper_kpa || missing_gas)
+			overlays += "over_alert3"
+		if(harvest)
+			overlays += "over_harvest3"
+
+	// Update bioluminescence.
+	if(seed)
+		if(seed.biolum)
+			set_light(round(seed.potency/10))
+			if(seed.biolum_colour)
+				light_color = seed.biolum_colour
+			else
+				light_color = null
+			return
+
+	set_light(0)
+	return
+
+/obj/machinery/portable_atmospherics/hydroponics/proc/check_level_sanity()
+	//writepanic("[__FILE__].[__LINE__] ([src.type])([usr ? usr.ckey : ""])  \\/obj/machinery/portable_atmospherics/hydroponics/proc/check_level_sanity() called tick#: [world.time]")
+	//Make sure various values are sane.
+	if(seed)
+		health =     max(0,min(seed.endurance,health))
+	else
+		health = 0
+		dead = 0
+
+	mutation_level = max(0,min(mutation_level,100))
```

Comment:
Use `Clamp()`:

`mutation_level = Clamp(mutation_level, 100, 0)` - Yes that makes no sense, see side note.

And do this for the below code too.

**Side note:** Is the `max()` even needed, or do you have the numbers reverse, because as it stands it appears like this'll make sure it's `> 100`, then make sure it's `> 0`, **the latter of which is never useful.**